pkg/cache/metrics.go

```diff
@@ -0,0 +1,56 @@
+package cache
+
+import (
+	"context"
+	"time"
+
+	"github.com/rancher/lasso/pkg/metrics"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	cacheMetricsCollectionPeriod = 1 * time.Minute
+)
+
+func (f *sharedCacheFactory) collectMetrics() sharedCacheFactoryMetrics {
+	f.lock.RLock()
+	defer f.lock.RUnlock()
+
+	gvks := make(map[schema.GroupVersionKind]int)
+	for gvk, c := range f.caches {
+		items := c.GetStore().List()
+		gvks[gvk] = len(items)
+	}
+	return sharedCacheFactoryMetrics{
+		gvks: gvks,
+	}
+}
+
+type sharedCacheFactoryMetrics struct {
+	// gvks is the total count of cache items by GroupVersionKind
+	gvks map[schema.GroupVersionKind]int
+}
+
+func (f *sharedCacheFactory) startMetricsCollection(ctx context.Context) {
+	go func() {
+		contextID := metrics.ContextID(ctx)
+		timer := time.NewTimer(cacheMetricsCollectionPeriod)
+		defer timer.Stop()
+		for {
+			factoryMetrics := f.collectMetrics()
+			for gvk, count := range factoryMetrics.gvks {
+				metrics.IncTotalCachedObjects(contextID, gvk.Group, gvk.Version, gvk.Kind, float64(count))
```

Comment:
I was trying to keep things as similar to how they were as possible, but since I like those suggestions and I'm already modifying the signature anyways, I'll change it, thanks.